src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.CreateDefaultHandler.Android.cs

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Reflection;
+
+namespace System.Net.Http
+{
+    public partial class HttpClient
+    {
+        private const string CustomHandlerTypeEnvVar = "XA_HTTP_CLIENT_HANDLER_TYPE";
+
+        private static Type s_customHandlerType;
+
+        private static HttpMessageHandler CreateDefaultHandler()
+        {
+            if (s_customHandlerType != null)
+            {
+                return (HttpMessageHandler)Activator.CreateInstance(s_customHandlerType);
+            }
+
+            string envVar = Environment.GetEnvironmentVariable(CustomHandlerTypeEnvVar)?.Trim();
```

Comment:
Should this be cached in a static readonly? (to avoid the repeated lookup of the env-variable).

Reply:
in my view the code flow will mostly take the first fast-out check under `s_customHandlerType` because the Android http handler is expected to be default (to rely on OS for security, settings and features like http3, etc).

And in general we try to avoid `static readonly` for mobiles - it's not AOT friendly and affects binary size because of static initialization.

Reply:
Makes sense, and thanks for the info about AOT!

Reply:
Still your advice makes sense if I'm wrong and the path with SocketHttpHandler will be taken mostly 🙂 (AFAIK it's a current XA behavior) but I think it makes sense to make Android one to be default.